unstract/connectors/src/unstract/connectors/filesystems/unstract_file_system.py

```diff
@@ -69,3 +70,15 @@ def get_fsspec_fs(self) -> AbstractFileSystem:
     def test_credentials(self) -> bool:
         """Override to test credentials for a connector."""
         pass
+
+    @staticmethod
+    def get_connector_root_dir(input_dir: str, **kwargs: Any) -> str:
+        """Override to get root dir of a connector."""
+        return f"{input_dir.strip('/')}/"
+
+    def create_dir_if_not_exists(self, input_dir: str) -> None:
```

Comment:
We can use the auto_mkdir method of fsspec here also.

Reply:
@harini-venkataraman we need to create dir if not exists. Let this code be